src/main/java/edu/phema/elm_to_omop/translate/CorelatedCriteriaTranslator.java

```diff
@@ -0,0 +1,83 @@
+package edu.phema.elm_to_omop.translate;
+
+import edu.phema.elm_to_omop.helper.CirceUtil;
+import edu.phema.elm_to_omop.translate.criteria.CriteriaTranslator;
+import edu.phema.elm_to_omop.translate.criteria.comparison.ComparisonExpressionTranslator;
+import edu.phema.elm_to_omop.translate.criteria.correlation.CorrelatedQueryTranslator;
+import edu.phema.elm_to_omop.translate.exception.PhemaNotImplementedException;
+import org.hl7.elm.r1.*;
+import org.ohdsi.circe.cohortdefinition.CorelatedCriteria;
+import org.ohdsi.circe.cohortdefinition.Occurrence;
+
+import java.util.List;
+
+/**
+ * Class responsible for generating CorelatedCriteria [sic]. CorelatedCriteria criteria are used for applying
+ * restrictions to Circe domain Criteria (e.g. ConditionOccurrence).
+ * <p>
+ * These restrictions include:
+ * <p>
+ * - start and end window restrictions relative to the index event (the containing Criteria, or the cohort entry event)
+ * - occurrence count (e.g. EXACTLY 3 procedures, AT_LEAST 1 condition)
+ * - visit restriction relative to the index event (the containing Criteria, or the cohort entry event)
+ * <p>
+ * Confusingly, a Criteria has field called CorelatedCriteria, which is of type CriteriaGroup.
+ */
+public class CorelatedCriteriaTranslator {
```

Comment:
Note for myself - we are accepting misspelling of Correlated as Corelated because that's what Circe has.  Any instance we refer to the CorelatedCriteria object, we will use the Circe spelling.  So don't make any comments about spelling... just let it go ☃️

Reply:
Yeah, the misspelling is unfortunately. I tried to adopt the convention that when something is named in reference to the misspelt class, I kept the misspelling. I do use the correct spelling in places when I'm referring generally to the concept of correlation, which might make things worse, I'm not sure.